### PR TITLE
[Feature]: MAE official

### PR DESCRIPTION
**Pull request by YuanLiuuuuuu**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: docs/en/model_zoo.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ All models and part of benchmark results are recorded below.
...
 | [MoCo v3](https://github.com/open-mmlab/mmselfsup/blob/master/configs/selfsup/mocov3/README.md)                    | [mocov3_vit-small-p16_32xb128-fp16-coslr-300e_in1k-224](https://github.com/open-mmlab/mmselfsup/blob/master/configs/selfsup/mocov3/mocov3_vit-small-p16_32xb128-fp16-coslr-300e_in1k-224.py) | [model](https://download.openmmlab.com/mmselfsup/moco/mocov3_vit-small-p16_32xb128-fp16-coslr-300e_in1k-224_20220127-e9332db2.pth) &#124; [log](https://download.openmmlab.com/mmselfsup/moco/mocov3_vit-small-p16_32xb128-fp16-coslr-300e_in1k-224_20220115_171742.log.json) |
+| [MAE](https://github.com/open-mmlab/mmselfsup/blob/master/configs/selfsup/mae/README.md)                           | [mae_vit-b-p16_8xb512-coslr-400e_in1k](https://github.com/open-mmlab/mmselfsup/blob/master/configs/selfsup/mocov3/mae_vit-b-p16_8xb512-coslr-400e_in1k.py)                                   | [model](https://download.openmmlab.com/mmselfsup/mae/mae_vit-base-p16_8xb512-coslr-400e_in1k-224_20220223-85be947b.pth) &#124; [log](https://download.openmmlab.com/mmselfsup/mae/mae_vit-base-p16_8xb512-coslr-300e_in1k-224_20220210_140925.log.json)                       |
```
Comment (fangyixiao18) [inline docs/en/model_zoo.md]: `mae_vit-b-p16_8xb512-coslr-400e_in1k` -> `mae_vit-base-p16_8xb512-coslr-400e_in1k`

File: docs/en/model_zoo.md [hunk 2]
```diff
@@ -51,6 +52,12 @@ If not specified, we use linear evaluation setting from [MoCo](http://openaccess
 | SwAV                | [swav_resnet50_8xb32-mcrop-2-6-coslr-200e_in1k-224-96](https://github.com/open-mmlab/mmselfsup/blob/master/configs/selfsup/swav/swav_resnet50_8xb32-mcrop-2-6-coslr-200e_in1k-224-96.py)     | SwAV paper setting    | 70.55     |
 | MoCo v3             | [mocov3_vit-small-p16_32xb128-fp16-coslr-300e_in1k-224](https://github.com/open-mmlab/mmselfsup/blob/master/configs/selfsup/mocov3/mocov3_vit-small-p16_32xb128-fp16-coslr-300e_in1k-224.py) | MoCo v3 paper setting | 73.07     |
 
+
+### ImageNet Fine-tuning
+| Algorithm | Config                                                                                                                                                     | Remarks | Top-1 (%) |
+| --------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------- | ------- | --------- |
+| MAE       | [mae_vit-b-p16_8xb512-coslr-400e_in1k](https://github.com/open-mmlab/mmselfsup/blob/master/configs/selfsup/mocov3/mae_vit-b-p16_8xb512-coslr-400e_in1k.py) |         | 83.1      |
```
Comment (fangyixiao18) [inline docs/en/model_zoo.md]: the same name format

File: docs/zh_cn/model_zoo.md [hunk 3]
```diff
@@ -20,6 +20,7 @@
...
 | [MoCo v3](https://github.com/open-mmlab/mmselfsup/blob/master/configs/selfsup/mocov3/README.md)                    | [mocov3_vit-small-p16_32xb128-fp16-coslr-300e_in1k-224](https://github.com/open-mmlab/mmselfsup/blob/master/configs/selfsup/mocov3/mocov3_vit-small-p16_32xb128-fp16-coslr-300e_in1k-224.py) | [model](https://download.openmmlab.com/mmselfsup/moco/mocov3_vit-small-p16_32xb128-fp16-coslr-300e_in1k-224_20220127-e9332db2.pth) &#124; [log](https://download.openmmlab.com/mmselfsup/moco/mocov3_vit-small-p16_32xb128-fp16-coslr-300e_in1k-224_20220115_171742.log.json) |
+| [MAE](https://github.com/open-mmlab/mmselfsup/blob/master/configs/selfsup/mae/README.md)                           | [mae_vit-b-p16_8xb512-coslr-400e_in1k](https://github.com/open-mmlab/mmselfsup/blob/master/configs/selfsup/mocov3/mae_vit-b-p16_8xb512-coslr-400e_in1k.py)                                   | [model](https://download.openmmlab.com/mmselfsup/mae/mae_vit-base-p16_8xb512-coslr-400e_in1k-224_20220223-85be947b.pth) &#124; [log](https://download.openmmlab.com/mmselfsup/mae/mae_vit-base-p16_8xb512-coslr-300e_in1k-224_20220210_140925.log.json)                       |
```
Comment (fangyixiao18) [inline docs/zh_cn/model_zoo.md]: the same name format

**Review by fangyixiao18 — CHANGES_REQUESTED** [review]

**Review by fangyixiao18 — APPROVED** [review]

File: mmselfsup/core/optimizer/constructor.py [hunk 4]
```diff
@@ -23,6 +23,8 @@ class DefaultOptimizerConstructor:
                   lr, weight_decay, momentum, etc.
         paramwise_cfg (dict, optional): Parameter-wise options.
             Defaults to None
+        layer_decay (float): base value for layer wise learning rate decay.
```
Comment (ZwwWayne) [inline mmselfsup/core/optimizer/constructor.py]: It is better to build a new constructor to process this part

File: mmselfsup/models/backbones/mae_pretrain_vit.py [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+from ..builder import BACKBONES
+from ..utils import build_2d_sincos_position_embedding
+
+
+@BACKBONES.register_module()
+class MAEViT(VisionTransformer):
+    """Vision Transformer for MAE pre-training.
+
+    A PyTorch implement of : `An Image is Worth 16x16 Words: Transformers
```
Comment (ZwwWayne) [inline mmselfsup/models/backbones/mae_pretrain_vit.py]: of : -> of:

File: mmselfsup/models/heads/mae_head.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        target = self.patchify(x)
+        if self.norm_pix:
+            mean = target.mean(dim=-1, keepdim=True)
+            var = target.var(dim=-1, keepdim=True)
+            target = (target - mean) / (var + 1.e-6)**.5
+
+        loss = (pred - target)**2
+        loss = loss.mean(dim=-1)
+
+        loss = (loss * mask).sum() / mask.sum()
```
Comment (ZwwWayne) [inline mmselfsup/models/heads/mae_head.py:48]: We can use mask to select the pred and target first then calculate the loss to reduce the memory usage?

**Comment by ZwwWayne** [comment]
imgs_per_gpu -> samples_per_gpu

**Review by Jiahao000 — APPROVED** [review]
